app/layout.tsx

```diff
@@ -16,14 +17,16 @@ export default function RootLayout({
   return (
     <html lang="en">
       <body>
-        <TaskProvider>
-          <div className="flex justify-center py-5 bg-background h-screen overflow-y-auto">
-            <div className="flex flex-col w-1/2 gap-4">
-              <Menu />
-              {children}
+        <Provider>
+          <TaskProvider>
```

Comment:
Please review this task provider. It tries to fetch all tasks even if we are on login or signup page! My suggestion would be to remove it completely as there are built-in mechanisms in next js which do request caching.